blocks/graph.py

```diff
@@ -520,3 +521,82 @@ def apply_dropout(computation_graph, variables, drop_prob, rng=None,
         replacement.tag.replacement_of = variable
 
     return computation_graph.replace(replacements)
+
+
+def apply_batch_normalization(computation_graph, variables, gammas,
+                              betas, axis=0, epsilon=1e-7):
+    """Returns a graph to variables in a computational graph.
+
+    Parameters
+    ----------
+    computation_graph : instance of :class:`ComputationGraph`
+        The computation graph.
+    variables : list of :class:`~tensor.TensorVariable`
```

Comment:
Yeah, I wouldn't go so far as making assumptions either. I will read the paper myself after finishing my ICML reviewing stack (argh).